frontend/providers/ModalProvider.js

```diff
@@ -0,0 +1,24 @@
+import React, { createContext, useState } from 'react';
+
+const defaultModalContextValues = {
+  filter: undefined,
+  search: undefined
+};
+
+export const ModalContext = createContext(defaultModalContextValues);
+
+export const ModalContextProvider = ({ children }) => {
+  const [filter, setFilter] = useState(undefined);
+  const [search, setSearch] = useState(undefined);
+
```

Comment:
@VasylMartyniv  you can add data here as a state here for the modal (for before when allowing camera)